math/mathmore/src/GSLMultiRootFinder.cxx

```diff
@@ -158,7 +158,7 @@ const double * GSLMultiRootFinder::FVal() const {
 }
 const char * GSLMultiRootFinder::Name() const {
    // get GSL name
-   return (fSolver != 0) ? fSolver->Name().c_str() : "";
+   return (fSolver != 0) ? fSolver->Name().c_str() : nullptr;
```

Comment:
The compiler is [complaining](http://cdash.cern.ch/viewBuildError.php?type=1&buildid=362082) about it, so it could be a problem. I just noticed that the problem is not about the empty string, though, but about `fSolver->Name().c_str()`. So, in hindsight, while my "fix" is probably pointless, it does seem like the issue with returning a local address still remains.